File: ruby/hamming/hamming.rb
class Hamming
  
  def self.compute(strandA,strandB)
    
    if strandA.size != strandB.size
      raise ArgumentError.new("Strands should have the same size") 
    else
      dist = 0
      strandA.scan(/./).each_with_index do |nucleotide,index|
        dist+=1 if nucleotide != strandB[index]
      end
    end
    dist
  end

end

module BookKeeping
  VERSION = 3
end

File: javascript/hamming/hamming.js
var Hamming = function() { };

Hamming.prototype.compute = function(strandA,strandB) {
  if (strandA.length !== strandB.length) {
    throw new Error('DNA strands must be of equal length.');
  } else {
    var i = strandA.length;
    var distance = 0;
    while(i--){
      if (strandA[i] !== strandB[i]) { distance++; }
    }
    return distance;
  }
};

module.exports = Hamming;